### DISCUSSION TITLE
Conversion to PDF/A

### DISCUSSION THREAD

**Discussion by raimondozagami** [jgm/pandoc#10501] [Q&A]
Please, i look for a solution to easily convert to PDF/A.
I can use basic syntax of pandoc, but i'm not a software programmer.
I'd like to convert single files, multiple files and choose the versione of PDF/A to comply with.
Thankyou for your attention.
Raimondo Zagami

**Comment by jgm** [comment]
The manual has some information about generating PDF/A via ConTeXt.  See under "Variables for ConTeXt."

**Reply by raimondozagami** [reply]
Yes, i read it, but i'm not able to write the instructions. Please, could you write the instructions.
I need:
- conversion of a single file
- conversion of multiple files in a folder
- possibility to choose the version (for instance PDF/A 1.b)
- use it on mac and win
Raimondo Zagami

**Comment by jpcirrus** [comment]
Yes, it is possible with latex, but only when using the engines `lualatex` (preferred, and better supported) or `pdflatex`, and you have to use a custom template. That template can remain identical to `default.latex` except that the following needs to be inserted at the very beginning of the template (i.e., before `$passoptions.latex()$`):

```latex
\DocumentMetadata{% Activate PDF resource management
  lang=${if(lang)}${lang}${else}en-US${endif},% Change default as appropriate
  pdfversion=${if(pdfversion)}${pdfversion}${else}1.7${endif},% Change default as appropriate
  pdfstandard=${if(pdfstandard)}${pdfstandard}${else}A-3u${endif},% Change default as appropriate
  testphase=new-or-1% Change as appropriate (see documentation)
}
```

You also have to have the `pdfmanagement-testphase` package and it's dependencies installed (one of which I remember is `tagpdf`). All the information you require regarding appropriate settings can be found in the documentation of `pdfmanagement-testphase`, `hyperref-generic` (the `hyperref` driver that is now used), and another document which I can't find right now (EDIT: `documentmetadata-support-code`) which also provides the various values for the `testphase` key which specifies the level of [tagging support](https://latex3.github.io/tagging-project/) (the one above is minimal and works with most classes and packages).

Compliance of the generated PDF with the standard can be tested at [veraPDF](https://demo.verapdf.org/).

**Reply by fkohrt** [reply]
Another thing to note is that the title metadata need to be provided separately if the title contains commas, as they need to be escaped with curly braces:

```
---
title: "A, B, and C"
title-meta: "`A{,} B{,} and C`{=latex}"
lang: en-US
---

Hello, world!
```

Otherwise the title metadata only contain everything up to the first comma.

See also: [pdftitle from hyperref not working as before](https://tex.stackexchange.com/questions/719927/pdftitle-from-hyperref-not-working-as-before)

There seems to be no such problem when using the lua filter described in issue [#5409](https://github.com/jgm/pandoc/issues/5409#issuecomment-2907799580).

**Reply by jgm** [reply]
I will fix the LaTeX writer so it adds these braces around commas automatically in `title-meta`.

**Comment by iandol** [comment]
Typst just added support for PDF/A-2b using `--pdf-standard a-2b` — and they are working to expand their support to other PDF standards

https://typst.app/blog/2024/typst-0.12/

**Reply by jgm** [reply]
Great! I tried this with pandoc:
```
pandoc --pdf-engine=typst --pdf-engine-opt=--pdf-standard=a-2b MANUAL.txt -o m.pdf
```
and it worked (passed validation with VeraPDF for a-2b level).

**Reply by iandol** [reply]
Typst 0.13 (release candidate): `Added support for PDF/A-3b export`
